Source/swiftlint/Extensions/File+CompilerArguments.swift

```diff
@@ -0,0 +1,136 @@
+import Foundation
+import SourceKittenFramework
+
+extension File {
+    /// Extracts compiler arguments for this file from the specified compiler logs.
+    ///
+    /// - parameter compilerLogs: xcodebuild log output from a clean & successful build (not incremental).
+    ///
+    /// - returns: The compiler arguments for this file, or an empty array if none were found.
+    func compilerArguments(compilerLogs: String) -> [String] {
+        return path.flatMap { path in
+            return compileCommand(compilerLogs: compilerLogs, sourceFile: path)
+        } ?? []
+    }
+}
+
+// MARK: - Private
+
+#if os(Linux)
+private extension Scanner {
+    func scanString(string: String) -> String? {
+        return scanString(string)
+    }
+}
+#else
+private extension Scanner {
+    func scanUpToString(_ string: String) -> String? {
+        var result: NSString?
+        let success = scanUpTo(string, into: &result)
+        if success {
+            return result?.bridge()
+        }
+        return nil
+    }
+
+    func scanString(string: String) -> String? {
+        var result: NSString?
+        let success = scanString(string, into: &result)
+        if success {
+            return result?.bridge()
+        }
+        return nil
+    }
+}
+#endif
+
+private func compileCommand(compilerLogs: String, sourceFile: String) -> [String]? {
+    let escapedSourceFile = sourceFile.replacingOccurrences(of: " ", with: "\\ ")
+    guard compilerLogs.contains(escapedSourceFile) else {
+        return nil
+    }
+
+    var compileCommand: [String]?
+    compilerLogs.enumerateLines { line, stop in
```

Comment:
Maybe a future optimization, but I wonder if it would be worth creating a dictionary of file path to commands up front that you could more easily access. I have to assume that doing this over and over is expensive with how large these files can be?